src/App.tsx

```diff
@@ -1,26 +1,46 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
 import React from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoProvider, useTodos } from './utils/TodoContext';
+import { Errors } from './components/Errors/Errors';
+import { TodoContent } from './components/TodoContent/TodoContent';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/TodoList/TodoList';
+import { useClearCompleted } from './hooks/useClearCompleted';
 
-const USER_ID = 0;
+const AppContent: React.FC = () => {
+  const { todos, filteredTodos, error, setFilter, setError, filter } =
+    useTodos();
+  const { handleClearCompleted, error: clearCompletedError } =
+    useClearCompleted();
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
+  const combinedError = error || clearCompletedError;
+
+  if (!todos) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
+    <div className="todoapp">
+      <TodoContent onErrorChange={setError}>
+        <TodoList todos={filteredTodos} onErrorChange={setError} />
+        <Footer
+          todos={todos}
+          filter={filter}
+          onFilterChange={setFilter}
```

Comment:
you shouldn't pass state setters like setFilter and setError as props and if you're using Context you can get access these props directly inside your components instead of passing them